File: 206_Reverse_Linked_List.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    if (head === null) return null;
    
    var p = head, temp = null;
    while(p.next !== null)
    {
        head = p.next;
        p.next = temp;
        temp = p;
        p = head;
    }
    p.next = temp;
    
    return head;
};

File: 206_Reverse_Linked_List.c
#include <stdio.h>
struct ListNode {
  int val;
  struct ListNode *next;
};

struct ListNode* reverseList(struct ListNode* head) {
  if (head == NULL) return NULL;

  struct ListNode* p = head;
  struct ListNode* temp = NULL;
  while (p->next != NULL) {
    head = p->next;
    p->next = temp;
    temp = p;
    p = head;
  }
  p->next = temp;

  return head;
}

int main() {
  return 0;
}
